Framework/Algorithms/test/ConvertSpectrumAxis2Test.h

```diff
@@ -468,6 +468,25 @@ class ConvertSpectrumAxis2Test : public CxxTest::TestSuite {
       TS_ASSERT_DELTA((*axis)(i), double(i % 10), 1E-10);
     }
   }
+
+  void test_eventWS() {
+    const std::string outputWS("outWS");
+    const std::string target("theta");
+    auto testWS =
+        WorkspaceCreationHelper::createEventWorkspaceWithFullInstrument(500, 3);
+    Mantid::Algorithms::ConvertSpectrumAxis2 conv;
+    conv.setChild(true);
+    conv.initialize();
+
+    conv.setProperty("InputWorkspace", testWS);
+
+    conv.setPropertyValue("OutputWorkspace", outputWS);
+    conv.setPropertyValue("Target", target);
+    conv.execute();
+    TS_ASSERT(conv.isExecuted());
+    const MatrixWorkspace_sptr output = conv.getProperty("OutputWorkspace");
```

Comment:
Changes look good now, just one small thing: Can you add an assert here that tests whether this is actually an `EventWorkspace`?